app/services/permissions_checker.rb

```diff
@@ -1,17 +1,25 @@
 # frozen_string_literal: true
 
 class PermissionsChecker
-  def initialize(current_user:, permission_names:, user_id:, friendly_id:, record_id:)
+  def initialize(current_user:, permission_names:, user_id:, friendly_id:, record_id:, current_provider:)
     @current_user = current_user
     @permission_names = permission_names
     @user_id = user_id
     @friendly_id = friendly_id
     @record_id = record_id
+    @current_provider = current_provider
   end
 
   def call
+    # check to see if current_user has SuperAdmin role
     return true if @current_user.role == Role.find_by(name: 'SuperAdmin', provider: 'bn')
 
+    # check to see if current user is trying to access own provider info
+    return false if @current_user.provider != @current_provider
```

Comment:
Good catch!
That saves us a security ticket!!

Reply:
That shouldn't be a problem,
However, it may be better tp have this check as the first one even before the super admin check.